src/merge_iterator.rs

```diff
@@ -251,11 +273,22 @@ impl<T: KeyValueIterator> KeyValueIterator for MergeIterator<T> {
             if peeked_entry.key != current_kv.key {
                 break;
             }
-            if peeked_entry.seq > current_kv.seq {
+            if peeked_entry.seq > current_kv.seq
```

Comment:
similar comment to the above here about handling the case where the iterators return entries in reverse-seq-num order across iterators.